src/patients/ContactInfo.tsx

```diff
@@ -0,0 +1,211 @@
+import { Spinner, Row, Column, Icon } from '@hospitalrun/components'
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import SelectWithLabelFormGroup from '../components/input/SelectWithLableFormGroup'
+import TextFieldWithLabelFormGroup from '../components/input/TextFieldWithLabelFormGroup'
+import TextInputWithLabelFormGroup from '../components/input/TextInputWithLabelFormGroup'
+import { ContactInfoPiece } from '../model/ContactInformation'
+import ContactInfoTypes from './ContactInfoTypes'
+
+interface Props {
+  data: ContactInfoPiece[]
+  errors?: string[]
+  label: string
+  name: string
+  isEditable?: boolean
+  onChange?: (newData: ContactInfoPiece[]) => void
+  type: 'text' | 'email' | 'tel'
+  isValid?: (entry: string) => boolean
+  errorMessageLabel: string
+}
+
+const ContactInfo = (props: Props) => {
+  const {
+    data,
+    errors,
+    label,
+    name,
+    isEditable,
+    onChange,
+    type,
+    isValid,
+    errorMessageLabel,
+  } = props
+  const [tempErrors, setTempErrors] = useState<string[]>()
+
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    if (!onChange) {
+      return
+    }
+    if (data.length === 0) {
+      onChange([...data, { value: '' }])
+    }
+  }, [data, onChange])
+
+  const getError = (i: number) => {
+    const tempError = tempErrors ? tempErrors[i] : null
+    if (tempError) {
+      return t(tempError)
+    }
+
+    const error = errors ? errors[i] : null
+    if (error) {
+      return t(error)
+    }
+
+    return ''
+  }
+
+  const typeLabel = t('patient.contactInfoType.label')
+  const typeOptions = Object.values(ContactInfoTypes).map((value) => ({
+    label: t(`patient.contactInfoType.options.${value}`),
+    value: `${value}`,
+  }))
+
+  const addLabel = t('actions.add')
+
+  const onChangeValue = (event: any, prevValue: string) => {
+    if (!onChange) {
+      return
+    }
+
+    // eslint-disable-next-line no-shadow
+    const newData = data.map(({ value, type }) =>
+      value === prevValue ? { value: event.currentTarget.value, type } : { value, type },
+    )
+    onChange(newData)
+  }
+
+  const header = (
+    <Row className="header mb-2">
+      <Column xs={12} sm={4}>
+        <span className="">{typeLabel}</span>
+        <span className="d-sm-none"> &amp; {t(label)}</span>
+      </Column>
+      <Column className="d-none d-sm-block" sm={8}>
+        {t(label)}
+      </Column>
+    </Row>
+  )
+
+  // todo: acts strange when deleting empty rows above non-empty rows.
+  // suspect TextInputWithLabelFormGroup missing value
+  const entries = data.map((entry, i) => {
```

Comment:
note: might be a good idea to wait till the components library get updated, to see if that'll fix it